src/platform/packages/shared/kbn-data-grid-in-table-search/src/in_table_search_control.tsx

```diff
@@ -0,0 +1,217 @@
+      });
+    },
+    [getColumnIndexFromId, scrollToCell, onChangeCss, onChangeToExpectedPage, pageSize]
+  );
+
+  const {
+    matchesCount,
+    activeMatchPosition,
+    isProcessing,
+    goToPrevMatch,
+    goToNextMatch,
+    renderCellsShadowPortal,
+    resetState,
+  } = useFindMatches({ ...props, onScrollToActiveMatch });
+
+  const showInput = useCallback(() => {
+    setIsInputVisible(true);
+  }, [setIsInputVisible]);
+
+  const hideInput = useCallback(
+    (shouldReturnFocusToButton: boolean = false) => {
+      setIsInputVisible(false);
+      resetState();
+      shouldReturnFocusToButtonRef.current = shouldReturnFocusToButton;
+    },
+    [setIsInputVisible, resetState]
+  );
+
+  // listens for the cmd+f or ctrl+f keydown event to open the input
+  useEffect(() => {
+    const handleGlobalKeyDown = (event: KeyboardEvent) => {
+      if (
+        (event.metaKey || event.ctrlKey) &&
+        event.key === 'f' &&
+        shouldOverrideCmdF(event.target as HTMLElement)
+      ) {
+        event.preventDefault(); // prevent default browser find-in-page behavior
+        showInput();
+
+        // if the input was already open before, make sure to shift the focus back to it
+        (
+          containerRef.current?.querySelector(
+            `[data-test-subj="${INPUT_TEST_SUBJ}"]`
+          ) as HTMLInputElement
+        )?.focus();
+      }
+    };
+
+    document.addEventListener('keydown', handleGlobalKeyDown);
+
+    return () => {
+      document.removeEventListener('keydown', handleGlobalKeyDown);
+    };
+  }, [showInput, shouldOverrideCmdF]);
+
+  // returns focus to the button when the input was cancelled by pressing the escape key
+  useEffect(() => {
+    if (shouldReturnFocusToButtonRef.current && !isInputVisible) {
+      shouldReturnFocusToButtonRef.current = false;
+      (
+        containerRef.current?.querySelector(
+          `[data-test-subj="${BUTTON_TEST_SUBJ}"]`
+        ) as HTMLButtonElement
+      )?.focus();
+    }
+  }, [isInputVisible]);
+
+  return (
+    <div ref={(node) => (containerRef.current = node)} css={innerCss}>
+      {isInputVisible ? (
+        <>
+          <InTableSearchInput
+            matchesCount={matchesCount}
+            activeMatchPosition={activeMatchPosition}
+            isProcessing={isProcessing}
+            goToPrevMatch={goToPrevMatch}
+            goToNextMatch={goToNextMatch}
+            onChangeSearchTerm={onChange}
+            onHideInput={hideInput}
+          />
+          {/* We include it here so the same parent contexts (like KibanaRenderContextProvider, UnifiedDataTableContext etc) will be applied to the portal components too */}
+          {/* as they do for the current component */}
+          {renderCellsShadowPortal ? renderCellsShadowPortal() : null}
+        </>
+      ) : (
+        <EuiToolTip
+          content={i18n.translate('dataGridInTableSearch.inputPlaceholder', {
+            defaultMessage: 'Search in the table',
```

Comment:
```suggestion
            defaultMessage: 'Find in table',
```
Suggesting this wording as it's really more about finding like a ctrl + F rather than a "search" that would filter results based on the input.

Reply:
Thanks, @florent-leborgne!